app/src/main/java/org/mozilla/fenix/components/Core.kt

```diff
@@ -165,10 +167,23 @@ class Core(private val context: Context) {
                 ).canonicalPath
             )
         ) {
-            CompletableDeferred("very-insecure-key")
+            CompletableDeferred(passwordsEncryptionKey)
         }
     }
 
+    /**
+     * Shared Preferences that encrypt/decrypt using Android KeyStore and lib-dataprotect for 23+
+     * otherwise simply stored
+     */
+    val secureAbove22Preferences by lazy {
+        SecureAbove22Preferences(context)
+    }
+
+    private val passwordsEncryptionKey =
+        secureAbove22Preferences.getString(PASSWORDS_KEY) ?: generateEncryptionKey(KEY_STRENGTH).also {
+            secureAbove22Preferences.putString(PASSWORDS_KEY, it)
```

Comment:
I think we should keep track of cases where we generated a key, and then it disappeared for some reason. At least to see if this happens at all in the wild.

You can store a flag in regular preferences when you generated a key, and see if that flag is set when you see a `null` key.

We can then keep track of these events in sentry,
```
class PasswordStorageKeyRegeneratedException : Exception()
crashReporter.submitCaughtException(PasswordStorageKeyRegeneratedException())
```

Reply:
Do you feel strongly about the exception thing? I just made it a Sentry "capture"

Reply:
That should be fine, as long as we can keep track of it in sentry after.